server.js

```diff
@@ -1,11 +1,148 @@
 'use strict';
 
-const { ApolloServer } = require('apollo-server');
+require('dotenv').config();
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const session = require('express-session');
+const passport = require('passport');
+const saml = require('passport-saml');
+const { ApolloServer } = require('apollo-server-express');
+const { Op } = require('sequelize');
+const db = require('./models');
 const { schema } = require('./graphql');
 
-const server = new ApolloServer({ schema });
+const PORT = 4000;
+const HOST = `http://localhost:${PORT}`;
+// TODO: Change this session secret
+const SESSION_SECRET = 'notsecure';
+
+// This is used to configure login/logout functionality, e.g. the entry point to our IDP, 
+// the login callback, etc.
+const samlStrategy = new saml.Strategy(
+  {
+    path: '/login/callback',
+    entryPoint: `https://uwbp-paramedics.us.auth0.com/samlp/${process.env.AUTH0_CLIENT_ID}`,
+    acceptedClockSkewMs: -1, // SAML assertion not yet valid fix
+    logoutUrl: 'https://uwbp-paramedics.us.auth0.com/v2/logout',
+    additionalLogoutParams: {
+      client_id: process.env.AUTH0_CLIENT_ID,
+      federated: '',
+      returnTo: 'http://localhost:4000/logout/callback',
+    },
+  },
+  (profile, done) => {
+    // TODO: This will need to be changed when we switch to ADFS
+    const email =
+      profile[
+        'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'
+      ];
+    db.user
+      .findOne({
+        where: { email: { [Op.like]: email } },
+      })
+      .then((matchingUser) => {
+        // Save the nameId and nameIDFormat for logout
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml = {};
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml.nameID = profile.nameID;
+        /* eslint-disable-next-line no-param-reassign */
+        matchingUser.saml.nameIDFormat = profile.nameIDFormat;
```

Comment:
is there any particular reason we want to mutate `matchingUser` instead of assembling a separate object with the fields we care about? your comment on line 45 suggests that we only care about those two fields, but i'm guessing we have other fields we would also care about (e.g. email) - it would be better to be super explicit and list out exactly what we want and assemble that object manually instead of relying on looking at the `matchingUser` object and hoping that has everything we need

(this would also eliminate having to disable these linter rules - as a general rule of thumb, if you ever see that you're disabling rules like this, maybe reconsider what you're trying to do and if there's a better way to accomplish the same result)

Reply:
```suggestion
        const returnVal = {
          user: matchingUser.
          saml: {
            nameID: profile.nameID,
            nameIDFormat: profile.nameIDFormat,
          }
        }
```

Reply:
good catch, made this change in the latest commit